include/spdk/bdev.h

```diff
@@ -145,6 +146,56 @@ enum spdk_bdev_reset_type {
 typedef spdk_event_fn spdk_bdev_io_completion_cb;
 typedef void (*spdk_bdev_io_get_rbuf_cb)(struct spdk_bdev_io *bdev_io);
 
+union spdk_critical_warning_state {
+	uint8_t		raw;
+
+	struct {
+		uint8_t	available_spare		: 1;
+		uint8_t	temperature		: 1;
+		uint8_t	device_reliability	: 1;
+		uint8_t	read_only		: 1;
+		uint8_t	volatile_memory_backup	: 1;
+		uint8_t	reserved		: 3;
+	} bits;
+};
+SPDK_STATIC_ASSERT(sizeof(union spdk_critical_warning_state) == 1, "Incorrect size");
+
+/**
+ * health report page
+ */
+struct __attribute__((packed)) spdk_health_report_page {
```

Comment:
This looks like it is just a copy of the NVMe health data - if that is the intent, it should just use the structures from nvme_spec.h directly.